server/src/stateManagement.mjs

```diff
@@ -447,19 +462,52 @@ function mergeCustomizer(objValue, srcValue) {
   }
 }
 
-function updateState(userId, newState) {
-  const userState = getState(userId);
+function updateState(userId, newState, scope = "") {
```

Comment:
We do not need to add a scope variable here. The userId used will provide us our scope. This is why I'd mentioned that the update functionalities do not need to be updated.

Someone wishing to update the ~abc or "global" scope would just set the userId to "~abc" or "global" when calling the API/CLI. That would then end up here as the userId

Reply:
After conversation there was a mis-understanding on my part. Resolving conversation.